apps/fz_http/lib/fz_http/saml/start_proxy.ex

```diff
@@ -0,0 +1,55 @@
+defmodule FzHttp.SAML.StartProxy do
+  @moduledoc """
+  This proxy starts Samly.Provider with proper configs
+  (after `FzHttp.Conf.Cache` has started)
+  """
+
+  alias FzHttp.Conf
+
+  def child_spec(arg) do
+    %{id: __MODULE__, start: {__MODULE__, :start_link, [arg]}}
+  end
+
+  def start_link(_) do
+    samly = Samly.Provider.start_link()
+
+    Application.fetch_env!(:samly, Samly.Provider)
+    |> set_service_provider()
+    |> set_identity_providers()
+    |> refresh()
+
+    samly
+  end
+
+  def set_service_provider(samly_configs) do
+    Keyword.put(samly_configs, :service_providers, [
+      %{
+        id: "firezone",
+        entity_id: "urn:firezone.dev:firezone-app",
+        certfile: Path.join(:code.priv_dir(:fz_http), "cert/selfsigned.pem"),
```

Comment:
We probably want this to be user-configurable right? Organizations using SAML in production probably want to use their own certs issued by their companies' CA

Reply:
Hmm yeah. Add a mount point, generate-saml-cert shell script can also generate to the mount location.